<launch>
  <arg name="tagimage" default="bebop/image_raw" />
  <arg name="tagcaminfo" default="bebop/camera_info" />

  <node pkg="apriltags_ros" type="apriltag_detector_node" name="apriltag_detector" output="screen">
    <!-- Remap topic required by the node to custom topics -->
    <remap from="image_rect" to="$(arg tagimage)" />
    <remap from="camera_info" to="$(arg tagcaminfo)" />

    <!-- Optional: Subscribe to the compressed stream-->
    <param name="image_transport" type="str" value="compressed" />
    <param name="sensor_frame_id" type="str" value="bebop/camera_optical" />

    <!-- Select the tag family: 16h5, 25h7, 25h9, 36h9, or 36h11(default) -->
    <param name="tag_family" type="str" value="36h11" />

    <!-- Enable projected optical measurements for more accurate tag transformations -->
    <!-- This exists for backwards compatability and should be left true for new setups -->
    <param name="projected_optics" type="bool" value="true" />

    <!-- Describe the tags -->
    <rosparam param="tag_descriptions">[
      {id: 0, size: 0.168},
      {id: 29, size: 0.168},
      {id: 39, size: 0.168},
      {id: 40, size: 0.168},
      {id: 41, size: 0.168},
      {id: 42, size: 0.168}
    ]
    </rosparam>
  </node>
</launch>
